packages/atomic/src/components/atomic-search-interface/atomic-search-interface.tsx

```diff
@@ -275,16 +276,43 @@ export class AtomicSearchInterface {
     return searchConfigFromProps;
   }
 
-  private getAnalyticsConfig(options: InitializationOptions) {
-    const analyticsConfigFromProps = {enabled: this.analytics};
+  private getAnalyticsConfig(
+    options: InitializationOptions
+  ): AnalyticsConfiguration {
+    const enabledFromProps = {
+      enabled: this.analytics,
+    };
     if (options.analytics) {
       return {
         ...options.analytics,
-        ...analyticsConfigFromProps,
+        ...enabledFromProps,
+        analyticsClientMiddleware: (event, payload) =>
+          this.augmentAnalyticsWithAtomicVersion(
+            event,
+            payload,
+            options.analytics?.analyticsClientMiddleware
+          ),
       };
     }
 
-    return analyticsConfigFromProps;
+    return {
+      ...enabledFromProps,
+      analyticsClientMiddleware: this.augmentAnalyticsWithAtomicVersion,
+    };
+  }
+
+  private augmentAnalyticsWithAtomicVersion(
```

Comment:
could we extract to reduce the number of line is this file? maybe `atomicVersionAnalyticsMiddleware` into global/environrment and extract this instead

Reply:
Perhaps `getAnalyticsConfig` could be extracted too? Refactoring to one file in charge of getting analytics configuration from A-Z,